haystack/modeling/model/predictions.py

```diff
@@ -201,13 +201,18 @@ def _span_to_string(self, token_offsets: List[int], clear_text: str) -> Tuple[st
         # final_text can be an empty string if start_t points to the very final token of the passage
         # final_text can be a whitespace if there is a whitespace token in the text, e.g.,
         # if the original text contained multiple consecutive whitespaces
-        if len(final_text.strip()) > 0:
-            final_text = final_text.strip()
-        else:
+        cleaned_final_text = final_text.strip()
+        if not cleaned_final_text:
             return "", 0, 0
-        end_ch = int(start_ch + len(final_text))
 
-        return final_text, start_ch, end_ch
+        # Adjust the offsets in case of whitespace at the beginning of the answer
+        right_offset = len(final_text.rstrip()) != len(final_text)
```

Comment:
When I read `right_offset` for the first time I expected a type integer and it's also used as if it was an integer in the line with `start_ch = start_ch + right_offset`. However, here `len(final_text.rstrip()) != len(final_text)` returns a boolean. I am also wondering if right means correct or the opposite of left here? `final_text.rstrip()` is about the end and not the beginning of the answer so I am confused.

Reply:
Sorry, WIP :sweat_smile: That's totally a typo